### PR TITLE
Announcement: Qubes Tor Onion services available again

### PR DESCRIPTION
**Pull request by andrewdavidwong**
QubesOS/qubes-issues#4911

CC: @unman, @adrelanos

### CODE REVIEW

File: 2019-04-12-tor-onion-services-available-again.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+```
+Website:   www.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion
+Yum repo:  yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion
+Deb repo:  deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion
+ISOs:      iso.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion
+```
+
+Soon, you will be able to get the new, correct repo definitions just by
+updating dom0 and your TemplateVMs. However, if you can't wait, please
```
Comment (unman) [inline 2019-04-12-tor-onion-services-available-again.md]: It still isn't explained *what you need to do "if you cant wait". The Whonix page is just about switching to onion repositories that are already set up.

I suggest:

However, if you can't wait, edit the repository definitions to include the new onion addresses. 
You need to make sure that **every** line that contains an onion address uses the appropriate **new** address above.
In dom0, edit /etc/yum.repos.d/qubes-dom0.repo and /etc/yum.repos.d/qubes-templates.repo.
In Fedora templates, edit /etc/yum.repos.d/qubes-r4.repo.
Every onion address should look like this:
#baseurl http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/..........

In Debian templates, edit /etc/sources.list.d/qubes-r4.list and check the entries in the "Qubes Tor updates repositories" section.
A sample entry would look like this:
#deb [arch=amd64] http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm stretch main

You can then enable updates using the new onion repositories by commenting out the clearnet definitions, and removing the comment from the start of the lines containing the new onion definitions.
There are step-by-step instructions on how to do this in the Whonix wiki:

Comment (andrewdavidwong) [reply 2019-04-12-tor-onion-services-available-again.md]: Updated.

**Comment by andrewdavidwong** [comment]
@marmarek, @unman, @adrelanos: Can we publish this?

**Comment by pgerber** [comment]
@andrewdavidwong, GitHub shows no changed files. Could it be that you removed the old file when updating the post date and forgot to include the new fie?

**Comment by adrelanos** [comment]
Yes.

**Comment by unman** [comment]
> Whonix templates do not require any action; their onions addresses

Change the rogue "onions" to "onion", but otherwise it's good to go.

**Comment by marmarek** [comment]
FWIW, related updates are already in current-testing:
https://github.com/QubesOS/updates-status/issues/1026 - package for VM
https://github.com/QubesOS/updates-status/issues/1025 - package for dom0
https://github.com/QubesOS/updates-status/issues/1022 - salt formula automatically updating repositories, even if the user modified the file

As for the post itself, it suggests this is something that every user need to do, while in fact it only affect users who choose to use onion repositories (an opt-in feature). This is especially important to note in case user do not use whonix gateway for updates at all, because in that case, switching to onion addresses will break the updates completely.
Besides adding info who is affected, maybe it would be better to shift actual instructions into a doc page linked from the article? Then, keep only introduction, new urls and a link to instructions?

**Comment by andrewdavidwong** [comment]
> GitHub shows no changed files. Could it be that you removed the old file when updating the post date and forgot to include the new fie?

Thanks, yes. Should be fixed now.

> Change the rogue "onions" to "onion", but otherwise it's good to go.

Fixed.

**Comment by andrewdavidwong** [comment]
> As for the post itself, it suggests this is something that every user need to do, while in fact it only affect users who choose to use onion repositories (an opt-in feature). This is especially important to note in case user do not use whonix gateway for updates at all, because in that case, switching to onion addresses will break the updates completely.

Well, it says, "if you can't wait," so only people who really want to do this should be doing it, but I'll try to make this even clearer.

> Besides adding info who is affected, maybe it would be better to shift actual instructions into a doc page linked from the article? Then, keep only introduction, new urls and a link to instructions?

Since this is something that expect people to do only this one time (or if a similar event recurs, which we do not intend and which would, in any case, be a very rare event), I don't think it makes sense to maintain a permanent doc page for this. Rather, I think it makes sense to include the instructions in the post when we expect for them to be used only for this occurrence.

**Comment by andrewdavidwong** [comment]
> Well, it says, "if you can't wait," so only people who really want to do this should be doing it, but I'll try to make this even clearer.

Done. I've also added instructions regarding the UpdateVM and `qubes.UpdatesProxy`.

**Comment by unman** [comment]
The Debian instructions need to make it explicit (as in Fedora) that it is necessary to comment out the Main repository and uncomment the Qubes Tor lines:

2. Find the section titled `Qubes Tor updates repositories`.
3. Update every `.onion` address to `deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion`.
   The `Qubes Tor updates repositories` section should look like this:
   ```
   # Qubes Tor updates repositories
   # Main qubes updates repository
   deb [arch=amd64] http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm stretch main
   #deb-src http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm stretch main
   ```
4. Comment out every line that has a clearnet address (http://deb.qubes-os.org)
5. Uncomment the onion repositories that you want to use.
6. In dom0, ensure that the first non-comment line in

**Comment by andrewdavidwong** [comment]
The instructions have been further clarified.

**Comment by andrewdavidwong** [comment]
@marmarek, @unman: Good to go?

**Comment by unman** [comment]
On Tue, Apr 16, 2019 at 09:54:41PM -0700, Andrew David Wong wrote:
> @marmarek, @unman: Good to go?
> 
> -- 
Didnt think you were waiting on me.
Good by me. Good job.

**Pull request merged**